### PR TITLE
Added command prompt cheatsheet

### PR DESCRIPTION
**Pull request by gautamkrishnar**
**What does your Instant Answer do?**
It will display all basic command prompt commands and shortcut keys
**What is the data source for your Instant Answer? (Provide a link if possible)**
(n/a)

**Why did you choose this data source?**
(n/a)

**Are there any other alternative (better) data sources?**
(n/a)

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Geeks

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
https://duck.co/ideas/idea/6303/command-prompt-cmd-exe-cheatsheet

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
screenshot 432

-----
IA Page: https://duck.co/ia/view/windows_cmd_cheat_sheet

### CODE REVIEW

**Comment by gautamkrishnar** [comment]
@moollaza  I had created the idea page and commented that i will do it but someone else claimed it!
check the idea page:
https://duck.co/ideas/idea/6303/command-prompt-cmd-exe-cheatsheet
my duck.co username:gkr

**Comment by gautamkrishnar** [comment]
@moollaza any idea why build is failing?

File: share/goodie/cheat_sheets/json/command-prompt.json [hunk 1]
```diff
@@ -0,0 +1,215 @@
+{
+    "id": "cmd_cheat_sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/command-prompt.json]: let's change this to "windows_cmd_cheat_sheet"

File: share/goodie/cheat_sheets/json/command-prompt.json [hunk 2]
```diff
@@ -0,0 +1,215 @@
+{
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/command-prompt.json]: Let's change the name to `windows-command-prompt.json`

File: share/goodie/cheat_sheets/json/command-prompt.json [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    "id": "cmd_cheat_sheet",
+    "name": "Command Prompt",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/command-prompt.json]: I think this should be "Windows Command Prompt" -- there are many different prompts out there, so we should make it really clear.

**Comment by gautamkrishnar** [comment]
## Yahoooo.... :smile: 
# All checks have passed!

**Comment by gautamkrishnar** [comment]
Hello @moollaza have you forgotten about this! :smile:

File: share/goodie/cheat_sheets/json/windows-command-prompt.json [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+            {
+                "key": "[Ctrl] + [C]",
+                "val": "Abort current command/typing"
+            },
+            {
+                "key": "[eft Alt] + [Left ⇧] + [PrtScn]",
+                "val": "Toggle High Visibility screen mode"
+            },
+            {
+                "key": "Ctrl-Z",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/windows-command-prompt.json]: Should be `[Ctrl] [Z]`

File: share/goodie/cheat_sheets/json/windows-command-prompt.json [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+            {
+                "key": " [PgDn] ",
+                "val": "Scroll window Down"
+            },
+            {
+                "key": "[Ctrl] + [C]",
+                "val": "Abort current command/typing"
+            },
+            {
+                "key": "[eft Alt] + [Left ⇧] + [PrtScn]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/windows-command-prompt.json]: - typo: "eft"
- `Left ⇧` should be `Left Shift` -- It's a little hard to discern the "Shift" Arrow IMO
- There should be no `+` between any keys (ever! :smiling_imp: )

File: share/goodie/cheat_sheets/json/windows-command-prompt.json [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+            {
+                "key": " [End] ",
+                "val": "Move to end of line"
+            },
+            {
+                "key": " [PgUp] ",
+                "val": "Scroll window up"
+            },
+            {
+                "key": " [PgDn] ",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/windows-command-prompt.json]: Please remove the leading/trailing spaces and the `[]`. They're only necessary when there are 2+ keys (content wrapped in `[]`) inside the `key`

File: share/goodie/cheat_sheets/json/windows-command-prompt.json [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+            {
+                "key": "[Ctrl] + [←]",
+                "val": "Move one word to the left (backward)"
+            },
+            {
+                "key": "[Ctrl] + [→]",
+                "val": "Move one word to the right (forward)"
+            },
+            {
+                "key": " [⌫] ",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/windows-command-prompt.json]: I'm not sure what arrow this is but let's stick to words. We're only using the directional arrows for now as they're simple enough to identify for just about all users.

File: share/goodie/cheat_sheets/json/windows-command-prompt.json [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+            {
+                "key": " [ESC] ",
+                "val": "Clear command line"
+            },
+            {
+                "key": " [INSERT] ",
+                "val": "Toggle Insert/Overwrite"
+            },
+            {
+                "key": "[Ctrl] + [Home]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/windows-command-prompt.json]: Again, remove the `+`'s

File: share/goodie/cheat_sheets/json/windows-command-prompt.json [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+            {
+                "key": " [F8] ",
+                "val": "Move backwards through the command history, but only display commands matching the current text at the command prompt"
+            },
+            {
+                "key": " [F9] ",
+                "val": "Run a specific command from the command history"
+            },
+            {
+                "key": " [ESC] ",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/windows-command-prompt.json]: Remove the leading/trailing spaces and `[]`. Should be: `"key": "Esc",`

File: share/goodie/cheat_sheets/json/windows-command-prompt.json [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+            {
+                "key": "[↓] / [↑] ",
+                "val": "Scroll through history of typed commands"
+            },
+            {
+                "key": " [F1] ",
+                "val": "Print characters of the previous command one by one"
+            },
+            {
+                "key": "[F2] [z]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/windows-command-prompt.json]: Let's be consistent about the caseing of the single letters. `Z`

**Comment by moollaza** [comment]
@gautamkrishnar thanks a lot for contributing, we really appreciate it! I've left you some feedback that needs to be implemented. Once things are updated please let us know by leaving a comment, so we can review this again. If you have any questions, feel free to ask them here!

File: share/goodie/cheat_sheets/json/windows-command-prompt.json (hunk 8 above)
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/windows-command-prompt.json]: Added pluses to show the key combinations...
Their are some shortcuts that user need to press a key, release it and then press another key

**Comment by gautamkrishnar** [comment]
@moollaza Is everything okay now :smile:

File: share/goodie/cheat_sheets/json/windows-command-prompt.json (hunk 8 above)
Comment (moollaza) [inline share/goodie/cheat_sheets/json/windows-command-prompt.json]: @gautamkrishnar in those cases a comma (`,`) should be placed between the keys.

Checkout the docs to see examples and explanations of this: https://duck.co/duckduckhack/goodie_cheat_sheets#simultaneous-keys-eg-pressing-a-and-b-together

**Comment by moollaza** [comment]
>Is everything okay now

@gautamkrishnar almost there! A couple more fixes and this is all done. Let's remove the `+` between keys that are pressed simultaneously and put a `,` between any keys that are pressed consecutively.

**Comment by gautamkrishnar** [comment]
@moollaza :smile: :+1:

**Comment by gautamkrishnar** [comment]
Thanks :smile:

**Comment by moollaza** [comment]
@gautamkrishnar congrats! It's live: https://duckduckgo.com/?q=windows+command+prompt+cheat+sheet&ia=cheatsheet&iax=1

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Comment by gautamkrishnar** [comment]
:+1:

**Pull request merged**